proto/parca/query/v1alpha1/query.proto

```diff
@@ -256,10 +259,46 @@ message QueryRequest {
   optional SourceReference source_reference = 9;
 
   // which runtime frames to filter out, often interpreter frames like python or ruby are not super useful by default
-  optional RuntimeFilter runtime_filter = 10;
+  optional RuntimeFilter runtime_filter = 10 [deprecated = true];
 
   // invert_call_stack inverts the call stacks in the flamegraph
   optional bool invert_call_stack = 11;
+
+  // a varying set of filters to apply to the query request
+  Filters filters = 12;
+}
+
+// Filters is a set of filters to apply to the query request
+message Filters {
+  // filter is the set of filters to apply to the query request
+  oneof filter {
+    // stack_filter is a filter for filtering by stacks
+    StackFilter stack_filter = 1;
+
+    // frame_filter is a filter for filtering by frames
+    FrameFilter frame_filter = 2;
+  }
+}
+
+// StackFilter is a filter for filtering by stacks
+message StackFilter {
+  // filter is the list of stacks to filter by
+  repeated string filter = 1;
```

Comment:
This API should be the same kind of one of, with the only one being FunctionNameStackFilter